### ISSUE TITLE
The Mac OS X menu bar disappears at startup when the help-browser is docked.

### ISSUE THREAD

**Issue by bernardgeyer** [supercollider/supercollider#3756]
The menu bar is available only if the help browser exists and is detached.

This is annoying because by starting SuperCollider on double-clicking an .scd file, it is not possible to load the project by making "Evaluate File".
Some users that are only musiciens think that the program is broken, unless they think to click in the window.

This behavior happens in versions 3.7.2, 3.8.1 and 3.9.3, but not in 3.6.6.
The problem doesn't exist in Windows.

**Comment by telephon** [comment]
Strange, I don't see it here.

**Comment by bernardgeyer** [comment]
Of course, when you click in the window to select it, the menu appears again.
But not everyone try to do that. Anyway it's not a normal behavior.

I have seen it on 3 different computers (with french and german keyboards) with the 3 last versions of the operating systems.

**Comment by brianlheim** [comment]
@bernardgeyer Can you be clearer about what you mean? I do not see the menu bar disappearing, but I see this:

<img width="758" alt="image" src="https://user-images.githubusercontent.com/15369640/41130983-81603f62-6ab9-11e8-8e99-1cc24b4c638b.png">

Which is a result of the sclang process gaining focus as the active application.

I suppose it might be helpful for the IDE to request focus again after successfully booting sclang. I'm not sure what event we would need to wait for to ensure that works correctly.

**Comment by bernardgeyer** [comment]
Yes, there is a menu bar, but an empty menu bar.
This was not the case in 3.6

**Comment by jamshark70** [comment]
Brian actually suggested an explanation for the behavior: "Which is a result of the sclang process gaining focus as the active application."

That is, it's the result of differences between typical window managers and Linux (and Windows) and OSX's window manager.

On all three platforms, the procedure of launching SC is the same. But OSX displays separate processes differently from Linux and Windows.

1. Launch the IDE process.

2. The IDE runs the interpreter (sclang) process. (Later, optionally, sclang can launch the scsynth process.)
   - Linux window managers typically embed the menu bar into the application window. Also, sclang by default does not open any GUI windows, so the IDE window retains focus.
   - OSX splits the application menu bar out to the top of the screen, so that the menus you see depend on the process that currently has focus. Because sclang can open GUI windows later, it needs to have its own menu bar separate from the IDE app (an OSX requirement). At step 1, the IDE app has focus. At step 2, sclang takes focus -- and sclang's menu bar becomes visible, hiding the IDE menus.

As Brian said, the solution would be to have a signal from sclang to tell the IDE process to take focus again. But it isn't clear what that signal should be.
